src/main/java/cceclipseplugin/editor/DocumentManager.java

```diff
@@ -184,16 +184,20 @@ public void handleNotification(Notification n) {
 			}
 			return;
 		}
-
-		this.applyPatch(n.getResourceID(), filepath, patches);
-
+		
 		ProjectMetadata projMeta = PluginManager.getInstance().getMetadataManager()
-				.getProjectMetadata(ResourcesPlugin.getWorkspace().getRoot().getLocation().toString() + "/");
+				.getProjectMetadata(projectRootPath.toString());
+		
+		IPath fileRelativePathWithName = new Path(fileMeta.getFilePath());
+		IPath projectRelativePath = new Path('/' + projMeta.getName());
```

Comment:
Does this need to have a preceding / ? or will the IPath constructor add it automatically.